she/evolve.py

```
@@ -0,0 +1,328 @@
+        if not self.incident_id:
+            raise EvolveError("incident_id required")
+        if not self.sha:
+            raise EvolveError("sha required")
+        needed = set(self.required_gates)
+        if not DUAL_GATES.issubset(needed):
+            raise EvolveError("dual gates repo-gate + termux-smoke must be required")
+        if self.live or self.mutates_source or self.promotion_ready:
+            raise EvolveError("P0.9 planner cannot be live, mutating, or promotion-ready")
+        kinds = {h.kind for h in self.hypotheses}
+        if self.selected_kind and self.selected_kind not in kinds:
+            raise EvolveError("selected_kind must name a planned hypothesis")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)